tests/php/test_deprecation.php

```diff
@@ -86,6 +87,39 @@ function test_deprecated_defined_functions( $function ) {
 		$this->assertTrue( function_exists( $function ) );
 	}
 
+	/**
+	 * @dataProvider provider_deprecated_method_stubs
+	 */
+	function test_deprecated_method_smoke_test( $class, $method ) {
+		$class = new ReflectionClass( $class );
+		$method = $class->getMethod( $method );
+		$parameters = $method->getParameters();
+
+		$arguments = array();
+
+		// Generating enough parameters for the method call to successfully complete.
+		foreach ( $parameters as $parameter ) {
+			if( $parameter->isDefaultValueAvailable() ) {
+
+				// No more parameters needed to successfully call the method.
+				break;
+			}
+
+			$arguments[] = 'Bogus argument';
+		}
+
+		try {
+			$method->invokeArgs( null, $arguments );
+		} catch ( Error $e ) {
+			$this->fail( $class->getName() . '::' . $method->getName() . ' is throwing fatal errors.' );
+			return;
+		}
+
+		// Marking as skipped instead of artificially passing to account for any warnings or notices
+		// the methods might throw when called with bogus arguments.
+		$this->markTestSkipped();
```

Comment:
Do we really need skip tests at this point? It seems that if this line is reached - the test is passed

Reply:
I also think it would be good to remove the markTestSkipped(), unless doing so introduces a lot of noise. Would it still work if you suppressed it via [@](https://www.php.net/manual/en/language.operators.errorcontrol.php)?

Also, might it be worth adding a comment to the top of the test that notes that it passes by simply not failing, since ordinarily folks look for assertions in tests?

Reply:
I couldn't make it pass without skipping because a deprecated notice somehow slips through the error handler, hence the `markTestSkipped` call here. If y'all have any ideas on how to do that, please let me know.

Reply:
Yeah, I think this is a good, clean way to avoid getting all the notices as failures in the tests.